### PR TITLE
add menu json checkup to flare and doctor

### PR DESCRIPTION
**Pull request by James-Pickett**

### CODE REVIEW

**Review by RebeccaMahany — DISMISSED** [review]

File: ee/debug/checkups/desktop_menu.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+	if _, err := os.Stat(menuJsonPath); err != nil {
+		d.status = Failing
+		d.summary = fmt.Sprintf("failed to stat menu json: %s", err)
+		return nil
+	}
+
+	file, err := os.Open(menuJsonPath)
+	if err != nil {
+		d.summary = fmt.Sprintf("failed to open %s: %s", menuJsonPath, err)
```
Comment (zackattack01) [inline ee/debug/checkups/desktop_menu.go:36]: do we want to set a status for any of these other error cases?

File: ee/debug/checkups/checkups.go [hunk 2]
```diff
@@ -120,6 +120,7 @@ func checkupsFor(k types.Knapsack, target targetBits) []checkupInt {
 		{&osqDataCollector{k: k}, doctorSupported | flareSupported},
 		{&osqRestartCheckup{k: k}, doctorSupported | flareSupported},
 		{&uninstallHistoryCheckup{k: k}, flareSupported},
+		{&desktopMenu{k: k}, doctorSupported | flareSupported},
```
Comment (directionless) [inline ee/debug/checkups/checkups.go:123]: I'd skip doctor here. I can't really imagine it being meaningful

File: ee/debug/checkups/desktop_menu.go (hunk 1 above)
Comment (James-Pickett) [reply ee/debug/checkups/desktop_menu.go:37]: yep, good catch, corrected ... set to failing at top, passing just before finishing

File: ee/debug/checkups/desktop_menu.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	menuJson, err := io.ReadAll(file)
+	if err != nil {
+		d.summary = fmt.Sprintf("failed to read menu.json: %s", err)
+		return nil
+	}
+
+	// just unmarshall the file to make sure it's valid json
+	var unmarshalledMenuJson any
+	if err := json.Unmarshal(menuJson, &unmarshalledMenuJson); err != nil {
+		d.summary = fmt.Sprintf("failed to unmarshal menu.json: %s", err)
```
[the remark this answers is not in the document]
Comment (James-Pickett) [reply ee/debug/checkups/desktop_menu.go]: good point, now writing the raw menu.json to file asap (forgoing pretty printing)

File: ee/debug/checkups/checkups.go (hunk 2 above)
Comment (James-Pickett) [reply ee/debug/checkups/checkups.go]: done

**Review by zackattack01 — APPROVED** [review]
nice!

File: ee/debug/checkups/desktop_menu.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (directionless) [reply ee/debug/checkups/desktop_menu.go]: yeah, no need to pretty print

**Pull request merged**